### ISSUE TITLE
fix(cognitive): Rust LoopExpression missing from cognitive nesting — loop {} scores zero complexity

### ISSUE THREAD

**Issue by dekobon** [dekobon/big-code-analysis#389]
## Summary

Rust's `loop {}` construct is invisible to the cognitive complexity metric. It receives zero structural increment and zero nesting contribution, while the structurally identical `for` and `while` loops are correctly handled.

## Location

- `src/metrics/cognitive.rs:436`

## Evidence

The nesting-increment arm at line 436 matches `ForExpression | WhileExpression | MatchExpression` but omits `LoopExpression`.

Empirical test (identical structure, different loop keyword):

| Function | Body | Cognitive | Cyclomatic |
|---|---|---|---|
| `test_for` | `for x in 0..10 { if x > 5 { break; } }` | **3** | 3 |
| `test_loop` | `loop { if true { break; } }` | **1** | 3 |
| `test_while` | `while true { if true { break; } }` | **3** | 3 |
| `test_deep_for` | `for { for { if { break } } }` | **6** | 4 |
| `test_deep_loop` | `loop { loop { if { break } } }` | **1** | 4 |

Cyclomatic complexity is correct in all cases (the `Loop` keyword token is correctly matched at `cyclomatic.rs:398`). Only cognitive is affected because it dispatches on expression-level node kinds, not keyword tokens.

## Expected Behavior

`test_loop` should report cognitive=3 (matching `test_for` and `test_while`). `test_deep_loop` should report cognitive=6.

## Actual Behavior

`test_loop` reports cognitive=1 (only the `if` contributes; the enclosing `loop` adds no structural increment and no nesting). `test_deep_loop` reports cognitive=1.

## Impact

Every `loop {}` in Rust code — retry loops, event loops, state machines, game loops — silently underreports cognitive complexity. Deeply nested `loop` blocks are scored as if they were flat. This makes cognitive thresholds ineffective for `loop`-heavy code (common in embedded, async runtimes, and systems programming).

## Root Cause

`src/metrics/cognitive.rs:436` lists the nesting-increment constructs but omits `LoopExpression`:

```rust
ForExpression | WhileExpression | MatchExpression => {
    increase_nesting(stats, &mut nesting, depth, lambda);
}
```

No other language is affected — Rust is the only supported language with a dedicated `loop` node kind separate from `while`. Other languages express infinite loops as `while(true)` which routes through `WhileExpression`/`WhileStatement`.

---

## Resolution

Fixed on branch `fix/issue-389` (commit aae5d40). Added
`LoopExpression` to the Rust cognitive nesting-increment arm at
`src/metrics/cognitive.rs:436` alongside `ForExpression`,
`WhileExpression`, and `MatchExpression`. Two anchored regression
tests added (`rust_loop_single`, `rust_loop_nested`) with integer
`assert_eq!` on `cognitive_sum()` in addition to JSON snapshots.
One integration snapshot in the `big-code-analysis-output`
submodule (`serde/src/ser/impls.rs.snap`) shifted by a pure
metric-value bump and was accepted. The `Fixes #389` trailer on
the commit will close this issue on merge.

**Comment by dekobon** [comment]
## General Plan for Fix

### Root Cause

`src/metrics/cognitive.rs:436` lists `ForExpression | WhileExpression | MatchExpression` as nesting-increment constructs but omits `LoopExpression`. The cyclomatic impl correctly handles `loop` via the `Loop` keyword token (line 398), but the cognitive impl dispatches on expression-level nodes.

### Implementation Steps

1. Add `| LoopExpression` to the nesting-increment arm at `src/metrics/cognitive.rs:436`:
   ```rust
   ForExpression | WhileExpression | LoopExpression | MatchExpression => {
       increase_nesting(stats, &mut nesting, depth, lambda);
   }
   ```

2. Accept snapshot changes (metric values will shift for any Rust file containing `loop`).

### Tests to Add/Update

- Add a test `rust_loop_cognitive_nesting` that asserts `loop { if true { break; } }` scores cognitive=3 (matching `for`/`while`)
- Add a test `rust_nested_loop_cognitive` that asserts `loop { loop { if true { break; } } }` scores cognitive=6
- Both tests should use `assert_eq!` on the headline value before the snapshot (per snapshot anchor policy)

### Verification

- [ ] `cargo test --workspace --all-features` passes
- [ ] `cargo clippy --workspace --all-targets -- -D warnings` clean
- [ ] `bca metrics` on the test file shows `loop` and `for` scoring identically
- [ ] Integration snapshots under `tests/repositories/big-code-analysis-output/` updated if affected

**Comment by dekobon** [comment]
Fixed on branch `fix/issue-389` (commit aae5d40).

## What changed

`src/metrics/cognitive.rs:436` — added `LoopExpression` to the
Rust nesting-increment arm alongside `ForExpression`,
`WhileExpression`, and `MatchExpression`. Before this, every
`loop {}` block was silently swallowed by the `_ => {}` catch-all
and contributed neither a structural `+1` nor a nesting bump.

Cyclomatic complexity already handled this correctly via the
`Loop` keyword token in `cyclomatic.rs`, so the asymmetry was
cognitive-only.

## Regression tests

Two anchored tests added to `cognitive.rs`:

- `rust_loop_single` — `loop { if true { break; } }` → cognitive=3
  (loop +1, nested `if` +2 at nesting depth 1).
- `rust_loop_nested` — `loop { loop { if true { break; } } }` →
  cognitive=6 (outer loop +1, inner loop +2, inner `if` +3).

Both use exact integer `assert_eq!` on `cognitive_sum()` in
addition to the JSON snapshot, per the project snapshot-anchoring
rule.

## Submodule impact

One integration snapshot in
`tests/repositories/big-code-analysis-output` shifted:
`serde/src/ser/impls.rs.snap` — a serde function using `loop {}`
moved from cognitive=1 to cognitive=3, and the file aggregate
from sum=34/avg=0.895 to sum=36/avg=0.947. Pure metric-value
shift, no structural drift. Accepted with `cargo insta test
--accept` and committed in the submodule.
